typed-racket-lib/typed-racket/env/env-utils.rkt

```diff
@@ -1,18 +1,20 @@
 #lang racket/base
 
-(require racket/dict racket/sequence)
-(provide id< sorted-dict-map sorted-dict-for-each in-sorted-dict)
+(require "../utils/utils.rkt" racket/private/dict)
+(provide id< sorted-dict-map sorted-dict-for-each in-sorted-dict in-sorted-dict-keys)
 
 (define (id< a b) (symbol<? (syntax-e a) (syntax-e b)))
 
-(define (sorted-dict-map dict f <)
-  (define sorted (sort (dict-map dict cons) (λ (x y) (< (car x) (car y)))))
-  (map (lambda (a) (f (car a) (cdr a))) sorted))
+(define-syntax-rule (in-sorted-dict dict <?)
+  (in-assoc (sort (dict->list dict) (λ (entry1 entry2) (<? (car entry1) (car entry2))))))
```

Comment:
Would it be better to`(sort .... <? #:key car)`?

Reply:
In past benchmarking w/ sort, I found not using `#:key` and just providing a richer comparison function to always be faster... so you won't find `#:key` in any of my TR code.